### PR TITLE
ddl: add logs to backfill worker and refactor some methods

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/35983

Problem Summary:

### What is changed and how it works?

- Enable linter for ddl/ingest
- Add logs to the backfill worker
- Add `IndexPrefix` to tables.Table
- Tiny code refactor

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/reorg.go [hunk 1]
```diff
@@ -239,7 +239,9 @@ func (w *worker) runReorgJob(rh *reorgHandler, reorgInfo *reorgInfo, tblInfo *mo
 		}
 		rowCount, _, _ := rc.getRowCountAndKey()
 		logutil.BgLogger().Info("[ddl] run reorg job done", zap.Int64("handled rows", rowCount))
-		job.SetRowCount(rowCount)
+		if job.RowCount == 0 {
```
Comment (Defined2014) [inline ddl/reorg.go:242]: Why add this `if`?

File: ddl/reorg.go [hunk 2]
```diff
@@ -268,8 +270,9 @@ func (w *worker) runReorgJob(rh *reorgHandler, reorgInfo *reorgInfo, tblInfo *mo
 		return dbterror.ErrWaitReorgTimeout
 	case <-time.After(waitTimeout):
 		rowCount, doneKey, currentElement := rc.getRowCountAndKey()
-		// Update a job's RowCount.
-		job.SetRowCount(rowCount)
+		if job.RowCount == 0 {
```
Comment (Defined2014) [inline ddl/reorg.go:273]: ditto

File: ddl/reorg.go (hunk 1 above)
Comment (tangenta) [reply ddl/reorg.go]: Reverted.

**Review by zimulala — APPROVED** [review]
LGTM

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**